pubsub/consumer.go

```diff
@@ -0,0 +1,152 @@
+package pubsub
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/go-redis/redis/v8"
+	"github.com/google/uuid"
+	"github.com/offchainlabs/nitro/util/redisutil"
+	"github.com/offchainlabs/nitro/util/stopwaiter"
+	"github.com/spf13/pflag"
+)
+
+type ConsumerConfig struct {
+	// Intervals in which consumer will update heartbeat.
+	KeepAliveInterval time.Duration `koanf:"keepalive-interval"`
+	// Duration after which consumer is considered to be dead if heartbeat
+	// is not updated.
+	KeepAliveTimeout time.Duration `koanf:"keepalive-timeout"`
+	// Redis url for Redis streams and locks.
+	RedisURL string `koanf:"redis-url"`
+	// Redis stream name.
+	RedisStream string `koanf:"redis-stream"`
+	// Redis consumer group name.
+	RedisGroup string `koanf:"redis-group"`
+}
+
+func ConsumerConfigAddOptions(prefix string, f *pflag.FlagSet, cfg *ConsumerConfig) {
+	f.Duration(prefix+".keepalive-interval", 30*time.Second, "interval in which consumer will perform heartbeat")
+	f.Duration(prefix+".keepalive-timeout", 5*time.Minute, "timeout after which consumer is considered inactive if heartbeat wasn't performed")
+	f.String(prefix+".redis-url", "", "redis url for redis stream")
+	f.String(prefix+".redis-stream", "default", "redis stream name to read from")
+	f.String(prefix+".redis-group", defaultGroup, "redis stream consumer group name")
+}
```

Comment:
See what we do with most configs:
We declare two variables of type ConsumerConfig:
*DefaultConsumerConfig that holds the default, and AddOptions reads default values from it
*TestConsumerConfig is used for tests with much lower values